src/extractor/guidance/turn_handler.cpp

```diff
@@ -470,125 +527,133 @@ Intersection TurnHandler::assignRightTurns(const EdgeID via_edge,
     return intersection;
 }
 
-std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
-                                                          const Intersection &intersection) const
+// finds a fork candidate by just looking at the geometry and angle of an intersection
+boost::optional<TurnHandler::Fork>
+TurnHandler::findForkCandidatesByGeometry(Intersection &intersection) const
```

Comment:
Oh, maybe you can help me here cause I was having trouble with this part anyways: If I make intersection const, all iterators I can get are const-iterators. The function is supposed to return a Fork struct with iterators to the beginning and end of the fork end those iterators cannot be const-iterators cause later on I want to modify their instructions. The only solution I got working was to make the intersection non-const, thus resulting in that code.

you have any idea how to fix that?